### PR TITLE
Postgres datastore errors and cleanup

### PR DESCRIPTION
**Pull request by jmank88**
This change picks a few small things from #523, and improves some additional error handling.

@martinpinto If it would be smoother to wait for #575 to merge first, or to open a PR against that branch, let me know.

### CODE REVIEW

File: api/datastore/postgres/postgres.go [hunk 1]
```diff
@@ -47,10 +47,6 @@ type rowScanner interface {
 	Scan(dest ...interface{}) error
 }
 
-type rowQuerier interface {
-	QueryRow(query string, args ...interface{}) *sql.Row
-}
-
```
Comment (jmank88) [inline api/datastore/postgres/postgres.go]: Unused

File: api/datastore/postgres/postgres.go [hunk 2]
```diff
@@ -88,14 +84,6 @@ func (ds *PostgresDatastore) InsertApp(ctx context.Context, app *models.App) (*m
...
 	var err error
 
-	if app == nil {
-		return nil, models.ErrDatastoreEmptyApp
-	}
-
-	if app.Name == "" {
-		return nil, models.ErrDatastoreEmptyAppName
-	}
-
```
Comment (jmank88) [inline api/datastore/postgres/postgres.go]: Redundant

File: api/datastore/postgres/postgres.go [hunk 3]
```diff
@@ -411,15 +401,18 @@ func scanRoute(scanner rowScanner, route *models.Route) error {
 		&headerStr,
 		&configStr,
 	)
+	if err != nil {
+		return err
+	}
 
 	if headerStr == "" {
 		return models.ErrRoutesNotFound
 	}
```
Comment (jmank88) [inline api/datastore/postgres/postgres.go]: Can anyone explain this one?  Why must the `headerStr` field be populated?  Why is this seemingly bad-data reported the same as a route not being found?  Note that it is not only called from `GetRoute` but `GetRoutes` as well, which does some questionable error discarding.

Comment (jmank88) [inline api/datastore/postgres/postgres.go]: Additionally, this `headerStr` check was happening *before* checking the `Scan` err.

File: api/datastore/postgres/postgres.go [hunk 4]
```diff
@@ -185,17 +173,18 @@ func (ds *PostgresDatastore) GetApp(ctx context.Context, name string) (*models.A
 	var resName string
 	var config string
 	err := row.Scan(&resName, &config)
+	if err != nil {
+		if err == sql.ErrNoRows {
+			return nil, models.ErrAppsNotFound
+		}
+		return nil, err
+	}
```
Comment (denismakogon) [inline api/datastore/postgres/postgres.go]: Shouldn't we include some logging here?

File: api/datastore/postgres/postgres.go [hunk 5]
```diff
@@ -209,10 +198,11 @@ func scanApp(scanner rowScanner, app *models.App) error {
 		&app.Name,
 		&configStr,
 	)
+	if err != nil {
+		return err
```
Comment (denismakogon) [inline api/datastore/postgres/postgres.go]: same here, it would be useful at least at DEBUG level.

File: api/datastore/postgres/postgres.go (hunk 4 above)
Comment (jmank88) [inline api/datastore/postgres/postgres.go]: I haven't looked everywhere, but generally the caller of `Datastore` methods handles the logging (with `api/server/error_response.go` or something similar).

Comment (jmank88) [inline api/datastore/postgres/postgres.go]: I would like to wrap most of the errors (the random, non-static ones) across the Datsatore with more context, but I didn't want to break local style too much.

**Pull request merged**